File: src/test/java/com/raychenon/leetcode/ContiguousArrayTest.java
package com.raychenon.leetcode;

import org.junit.BeforeClass;
import org.junit.Test;

import static junit.framework.TestCase.assertEquals;

/**
 * User: raychenon
 * Date: 15/4/2020
 * https://leetcode.com/problems/contiguous-array/
 */
public class ContiguousArrayTest {

    private static ContiguousArray instance;

    @BeforeClass
    public static void setUp() {
        instance = new ContiguousArray();
    }

    @Test
    public void example1Test() {
        assertEquals(2, instance.findMaxLength(new int[]{0, 1}));
    }

    @Test
    public void example2Test() {
        assertEquals(2, instance.findMaxLength(new int[]{0, 1, 0}));
    }

}

File: src/test/kotlin/com/raychenon/kotlin/leetcode/ContiguousArrayTest.kt
package com.raychenon.kotlin.leetcode

import junit.framework.TestCase
import org.junit.Test

/**
 * User: raychenon
 * Date: 16/4/2020
 * https://leetcode.com/problems/contiguous-array/
 */
class ContiguousArrayTest {

    @Test
    fun example1Test() {
        assertMethods(2, intArrayOf(0, 1))
    }

    @Test
    fun example2Test() {
        assertMethods(2, intArrayOf(0, 1, 0))
    }

    fun assertMethods(expected: Int, nums: IntArray) {
        TestCase.assertEquals(expected, ContiguousArray.findMaxLength(nums))
        TestCase.assertEquals(expected, ContiguousArray.findMaxLengthBruteForce(nums))
    }

}